cpp/src/arrow/compute/key_hash.cc

```diff
@@ -378,24 +378,43 @@ void Hashing32::HashFixed(int64_t hardware_flags, bool combine_hashes, uint32_t
   }
 }
 
-void Hashing32::HashMultiColumn(const std::vector<KeyColumnArray>& cols,
+Status Hashing32::HashMultiColumn(const std::vector<KeyColumnArray>& cols,
                                 LightContext* ctx, uint32_t* hashes) {
   uint32_t num_rows = static_cast<uint32_t>(cols[0].length());
 
   constexpr uint32_t max_batch_size = util::MiniBatch::kMiniBatchLength;
+  const auto alloc_batch_size = std::min(num_rows, max_batch_size);
```

Comment:
Maybe it can be more clear if we combine these two lines to `const uint32_t max_batch_size = std::min(num_rows, util::MiniBatch::kMiniBatchLength);`.